packages/wasm/src/wasm_package.rs

```diff
@@ -59,25 +59,25 @@ impl WasmPackage {
         wasm_bytes: Vec<u8>,
         file_reader: Arc<dyn FileReader>,
         manifest: Option<Vec<u8>>,
-    ) -> Result<Self, WrapperError> {
-        Ok(Self {
+    ) -> Self {
+        Self {
             file_reader: Arc::new(InMemoryFileReader::new(file_reader, None, Some(wasm_bytes))),
             manifest,
             wasm_module: Arc::new(Mutex::new(Some(WasmModule::Compiled(wasm_module)))),
-        })
+        }
     }
 
     pub fn from_wasm_module(
         wasm_module: WasmModule,
         wasm_bytes: Vec<u8>,
```

Comment:
shouldn't we pass here `wasm_module`? rather than `wasm_bytes` & `wasm_module`

Reply:
This is for the `get_file` method on the Wrapper trait. If the user does `get_file("wrap.wasm")` we need to be able to return the actual byte code.

Reply:
if i am not mistaken, the `wasm_package` doesn't implement the `Wrapper` trait (hence it doesn't have `get_file`). And when it is creating the wrapper, it is passing the `file_reader` which should take care of that right?

Reply:
The problem is that we cannot use the standard file_reader to read in memory bytes. Hence why we use the InMemory one. This method is just a faster way for the user to provide bytes instead of creating the in memory reader himself.
 ```rust
 pub fn from_wasm_module(
        wasm_module: WasmModule,
        wasm_bytes: Vec<u8>,
        file_reader: Arc<dyn FileReader>,
        manifest: Option<Vec<u8>>,
    ) -> Self {
        Self {
            file_reader: Arc::new(InMemoryFileReader::new(file_reader, None, Some(wasm_bytes))),
            manifest,
            wasm_module: Arc::new(Mutex::new(Some(wasm_module))),
        }
    }
    ```

Reply:
gotcha, i feel we have some mix of concepts here

> The problem is that we cannot use the standard file_reader to read in memory bytes. Hence why we use the InMemory one.

iiuc, this means that file reader is not really a file reader but rather a "cache" of the wasm bytes and that it can be used to read other files :thinking:

anyways, this is an improvement so let's not waste time on this (for now :yum:)

Reply:
yea, the file reader reads ALL wrap files: wrap.info, wrap.wasm, README.md.. etc